lib/pull_request.rb

```diff
@@ -93,6 +93,8 @@ def approve!
 
   def merge!
     GitHubClient.instance.merge_pull_request("alphagov/#{@api_response.base.repo.name}", @api_response.number)
+  rescue StandardError => e
```

Comment:
I'm not sure that rescuing `StandardError` is the best pattern. It will make any other error harder to spot and debug. I also realised that we don't have Sentry configured for this app. 
Could we specifically rescue `Octokit::MethodNotAllowed`
```suggestion
  rescue Octokit::MethodNotAllowed => e
```

Reply:
I'm also wondering why it was `#is_auto_mergeable?` returned `true`
https://github.com/alphagov/govuk-dependabot-merger/blob/main/lib/pull_request.rb#L35-L36

Reply:
I think what the error and 405 response are saying is that auto-merger approved the PR but the PR was not in a state that allowed merging when we called the endpoint. Scenarios I can think of is when a PR has merge conflicts or is being rebased against main. This is likely to happen when we are merging PRs one after another.
So probably something we should look into because it's not ideal that we had to merge it manually (maybe we could check the `mergeable` property https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#get-a-pull-request?).

I agree that we should be rescuing the GitHub API errors as we don't want loop to break when one PR can't be mergerd (the ). How about `Octokit::Error`, which covers both client and server side error? https://octokit.github.io/octokit.rb/Octokit/Error.html would 
```suggestion
  rescue Octokit::Error => e
```

Reply:
That's a good point, I've noticed that when merging one PR Dependabot will rebase the others, so we should find a solution for that. I think we could look into using the mergeable property or even having a delay between merges, if that isn't possible.

With regards to the exceptions, I was just following what we did in the [Seal](https://github.com/alphagov/seal/blob/main/lib/github_fetcher.rb). I don't think these errors break the merger in any way, I'm just making this change to keep the logs clean.

Reply:
I think I would rescue `Octokit::MethodNotAllowed` or `Octokit::Error` here as well (better to be overly specific than overly general)

If GitHub is giving us a 405 when we try to merge, it’s because the PR isn’t in a mergeable state (https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#merge-a-pull-request--status-codes), and if `is_auto_mergeable?` is returning `true` when it's not actually mergeable then that sounds like a bug to me. Is it worth capturing a card around fixing that?

Reply:
I believe `is_auto_mergeable?` != mergeable because the method is designed to check for our specific business requirements. Including a mergeable check might result in merging only one PR per repository per day since the others won't be in a mergeable state during the rebase and would fail the check. I'd suggest considering the addition of a delay between merging PRs in the same repo.

Regarding error handling, given that this is not a production app and we won't actively address any errors but simply print them out, I fail to see the significance of which one is being caught. It's simply a cosmetic issue in the logs that we're attempting to address. The app is working as intended, apart from the mergeability issue mentioned above. However, I am open to changing it to `Octokit::Error` to avoid additional discussions.